ssl/t1_lib.c

```diff
@@ -354,14 +385,14 @@ int tls1_set_groups(uint16_t **pext, size_t *pextlen,
     for (i = 0; i < ngroups; i++) {
         unsigned long idmask;
         uint16_t id;
-        /* TODO(TLS1.3): Convert for DH groups */
         id = tls1_nid2group_id(groups[i]);
-        idmask = 1L << id;
-        if (!id || (dup_list & idmask)) {
+        idmask = 1L << (id & 0x00FF);
+        dup_list = (id < sizeof(*dup_list)) ? &dup_list_egrp : &dup_list_dhgrp;
```

Comment:
This condition is incorrect. It should be either id < 8*sizeof(*dup_list) or id < 0x100. Perhaps the second one is more clear even though it contains a magic number. The question also is should there be asserts or errors if the masks are overflowed?

Reply:
Fixed